controllers/storagecluster/platform_detection.go

```diff
@@ -35,6 +46,13 @@ type Platform struct {
 func (p *Platform) GetPlatform(c client.Client) (configv1.PlatformType, error) {
 	// if 'platform' is already set just return it
 	if p.platform != "" {
+		if p.platform == configv1.IBMCloudPlatformType {
+			platformIBM, platErr := getActualIBMPlatformType(c)
+			if platErr != nil {
+				return "", fmt.Errorf("Error checking COS secret in IBMCloud: %v", platErr)
+			}
+			p.platform = platformIBM
+		}
```

Comment:
I believe you don't need this here, since anything set in `getPlatform()` below will persist here, and I am generally assuming that the underlying platform will not change without restarting the ocs-operator Pod.

Reply:
When I tested the change, it was returning from `if p.platform != ""` condition. So had to add the check here

 log results
 ```
 {"level":"info","ts":1615528098.4257863,"logger":"controllers.StorageCluster","msg":"platform already set Inside CephObjectStore:'IBMCloud'","Request.Namespace":"openshift-storage","Request.Name":"ocs-storagecluster"}
{"level":"info","ts":1615528098.425803,"logger":"controllers.StorageCluster","msg":"Inside CephObjectStore:'IBMCloud'","Request.Namespace":"openshift-storage","Request.Name":"ocs-storagecluster"}
{"level":"info","ts":1615528098.5261004,"logger":"controllers.StorageCluster","msg":"creating CephObjectStore ocs-storagecluster-cephobjectstore","Request.Namespace":"openshift-storage","Request.Name":"ocs-storagecluster"}
{"level":"info","ts":1615528098.6885338,"logger":"controllers.StorageCluster","msg":"Creating cephObjectStoreUser ocs-storagecluster-cephobjectstoreuser","Request.Namespace":"openshift-storage","Request.Name":"ocs-
```